core/templates/dev/head/pages/state_editor/StateHintsEditorDirective.js

```diff
@@ -250,6 +250,10 @@ oppia.directive('stateHintsEditor', [
                 StateHintsService.saveDisplayedValue();
                 $scope.onSaveHints(StateHintsService.displayed);
               }
+
+              if (index === $scope.activeHintIndex) {
```

Comment:
@abhaygupta97, Can we use `$scope.changeActiveHintIndex`? (as I can see that function used to handle all cases even the one you have updated right now!)
Let me know if it's not possible!

Reply:
We cannot use `$scope.changeActiveHintIndex` because it would fail in two cases:

1. Suppose no hint is active. On pressing delete hint if we call `$scope.changeActiveHintIndex(index)`, due to [else](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/pages/state_editor/StateHintsEditorDirective.js#L98) `$scope.activeHintIndex` would become equal to `index` which is the index of deleted hint.
2. Suppose there are two hints. One hint is active and we press delete hint on the non-active hint. On calling `$scope.changeActiveHintIndex(index)`, due to [else](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/pages/state_editor/StateHintsEditorDirective.js#L98) `$scope.activeHintIndex` would become equal to `index` which is the index of deleted hint.

Reply:
Thanks for looking into it!